packages/cli/src/fingerprint/fingerprintWatchCommand.ts

```diff
@@ -45,6 +45,19 @@ export default class FingerprintWatchCommand {
     }
   }
 
+  async watcherErrorFunction (error: Error) {
+    if (error.message.includes("ENOSPC: System limit for number of file watchers reached")) {
```

Comment:
We should check to see if the watcher was already removed before catching the error. This line could be:
```
if (this.watcher && error.message.includes("ENOSPC: System limit for number of file watchers reached")) {
```